unic-langid-impl/src/lib.rs

```diff
@@ -482,19 +484,20 @@ impl LanguageIdentifier {
     /// let li1: LanguageIdentifier = "ca-ES-valencia".parse()
     ///     .expect("Parsing failed.");
     ///
-    /// assert_eq!(li1.get_variants(), &["valencia"]);
+    /// assert_eq!(li1.get_variants().collect::<Vec<_>>(), &["valencia"]);
     ///
     /// let li2: LanguageIdentifier = "de".parse()
     ///     .expect("Parsing failed.");
     ///
     /// assert_eq!(li2.get_variants().len(), 0);
     /// ```
-    pub fn get_variants(&self) -> Vec<&str> {
-        if let Some(variants) = &self.variants {
-            variants.iter().map(|s| s.as_ref()).collect()
-        } else {
-            vec![]
-        }
+    pub fn get_variants(&self) -> impl ExactSizeIterator<Item = &str> {
+        let variants: &[_] = match self.variants {
```

Comment:
The `: &[_]` is not really needed, was just for clarity. Feel free to remove it if you wish.

Reply:
I'll keep it for readability.